include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -120,12 +108,27 @@ def AndOp : UTVariadicRTLOp<"and", [Commutative]>;
 def OrOp  : UTVariadicRTLOp<"or", [Commutative]>;
 def XorOp : UTVariadicRTLOp<"xor", [Commutative]>;
 
-def EQOp   : CmpBinRTLOp<"eq", [Commutative]>;
-def NEQOp  : CmpBinRTLOp<"neq", [Commutative]>;
-def LTOp   : CmpBinRTLOp<"lt">;
-def ULTOp  : CmpBinRTLOp<"ult">;
-def LEQOp  : CmpBinRTLOp<"leq">;
-def ULEQOp : CmpBinRTLOp<"uleq">;
+def ICmpPredicateEQ  : I64EnumAttrCase<"eq", 0>;
+def ICmpPredicateNE  : I64EnumAttrCase<"ne", 1>;
+def ICmpPredicateSLT : I64EnumAttrCase<"slt", 2>;
+def ICmpPredicateSLE : I64EnumAttrCase<"sle", 3>;
+def ICmpPredicateULT : I64EnumAttrCase<"ult", 4>;
+def ICmpPredicateULE : I64EnumAttrCase<"ule", 5>;
```

Comment:
This is ok, but is there a reason you're still deviating from what the std dialect is doing?

```
def CMPI_P_EQ  : I64EnumAttrCase<"eq", 0>;
def CMPI_P_NE  : I64EnumAttrCase<"ne", 1>;
def CMPI_P_SLT : I64EnumAttrCase<"slt", 2>;
def CMPI_P_SLE : I64EnumAttrCase<"sle", 3>;
def CMPI_P_SGT : I64EnumAttrCase<"sgt", 4>;
def CMPI_P_SGE : I64EnumAttrCase<"sge", 5>;
def CMPI_P_ULT : I64EnumAttrCase<"ult", 6>;
def CMPI_P_ULE : I64EnumAttrCase<"ule", 7>;
def CMPI_P_UGT : I64EnumAttrCase<"ugt", 8>;
def CMPI_P_UGE : I64EnumAttrCase<"uge", 9>;

def CmpIPredicateAttr : I64EnumAttr<
    "CmpIPredicate", "",
    [CMPI_P_EQ, CMPI_P_NE, CMPI_P_SLT, CMPI_P_SLE, CMPI_P_SGT,
     CMPI_P_SGE, CMPI_P_ULT, CMPI_P_ULE, CMPI_P_UGT, CMPI_P_UGE]> {
  let cppNamespace = "::mlir";
}
```

The advantage of that is that the canonicalization patterns can uniformly push constants to the RHS (by swapping the condition code).  Is this change something intentional?